packages/web-app-files/src/helpers/resource/resource.ts

```diff
@@ -1,5 +1,13 @@
 import { Resource } from 'web-client'
 import fileExtensions from '../extensions/fileExtensions'
+import { DavProperty } from 'web-pkg/src/constants'
+
+interface ExtractExtensionFromFileOptions {
+  name: string
+  type: string
+  isFolder: boolean
+  fileInfo: object
```

Comment:
huh, why don't you pass `fileInfo[DavProperty.Name]` as `name` prop into the input object?

Reply:
That's what I was trying to tell you earlier 😄  This method is called in `buildResource`, when the resource has not been built yet. Hence we don't have `fileInfo[DavProperty.Name]` as `name` yet - it just exists as `fileInfo` property at this point of time.

Reply:
but you can provide an object with `{ name: fileInfo[DavProperty.Name] }` instead of passing in the whole fileInfo struct.

Reply:
Ah okay, now I get it. Well then we can basically keep `extractExtensionFromFile` untouched and just adjust the argument passed in `buildResource`.